### DISCUSSION TITLE
Handlebars template for a small people pill to use on a card

### DISCUSSION THREAD

**Discussion by calebmil** [microsoft-search/pnp-modern-search#3553] [Q&A] [answered]
I am fairly new to handlebars and PnP Modern Search.  I am trying to build a Display List or Cards (preferred) handlebars template for a list item where the list definition includes one or more Person fields.  The effect I want to include is similar to the way the default Cards layout treats the Author field with a headshot, a name and a hover effect that shows the person card for that named user.

My questions are:

1. Where is the right place in the Search Results web part to define a handlebars template for a display card that aligns with my list item properties?  I thought this might be the results template but I do not see anything in the pre-provisioned handlebars that supports that assumption.
2. If making my own handlebars template is not the right approach, where can I define what the Managed Card fields are that I want to display?  I thought maybe Layout Slots but the card fields list seems to be a subset of my slots.  
3. How do I tell a card that the field I am entering is a Person field?  Is this done by defining as Results Type?
4. Lastly, does anyone have a good card Handlebars template that I can borrow from to achieve my goals?

**Comment by kasperbolarsen** [comment, marked as answer]
I haven't been able to create a hover enabled layout using just a Handlebar, I had to fiddle with a custom layout in order of it to work. The most "approachable" layout seem to be the List Layout: select a List layout, click in the "Edit results template", and add 
```
<mgt-person
                                        user-id="{{getUserEmail (slot item @root.slots.Author)}}"
                                        view='threelines'
                                        show-presence="true"
                                        person-card="hover"
                                        avatar-size="large"
                                        />
                                    </mgt-person>

```right after til Title section 
and you have a hover enabled mgt-person 


![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ed67d2a0-a2f4-446c-9cca-96d4a0ef2f17)

**Reply by calebmil** [reply]
Kasper,

It looks like you intended to post code but I am not seeing anything ...

Caleb

**Reply by kasperbolarsen** [reply]
how about now ? 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/d7d3caf3-7831-4d12-8ee0-97f02bcf9673)

**Reply by calebmil** [reply]
Much better.  Thank you!  I'll give this a try.  You're screen shot looks promising!

**Reply by calebmil** [reply]
Is this something that requires the latest release to work?  I'm on 4.9.3.  I can inspect the page and see the code being rendered with a user email listed as the user ID but no headshot or hover card is appearing in the UI ...

To confirm, I started with the layout set to "List" and clicked on the Edit Results Template link. I put that snippet inside the span for the Author like so:

```
<span class="template--listItem--author">
                                            
        <mgt-person
                                        user-id="{{getUserEmail (slot item @root.slots.Author)}}"
                                        view='threelines'
                                        show-presence="true"
                                        person-card="hover"
                                        avatar-size="large"
                                        />
        </mgt-person>

     {{#with (split (slot item @root.slots.Author) '|')}}
         {{[1]}}
     {{/with}}
</span>
```

 I also tried putting that block in between the span with Title and the span with Author but it made no visible difference in rendering.

**Reply by patrikhellgren** [reply]
That example should work with the version you are running.
Two things to check:

- Have you enabled _Use Microsoft Graph Toolkit_ in the Layout settings (Webpart properties page 2)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/31273837/29239843-72ee-4388-9622-dbebed5d5be7)

- Your issue could also appear if you have other SPFx solutions on the page using Microsoft Graph Toolkit (MGT) but another version. If so, there are various things to try but none of them works perfectly. This will be resolved in the next version of PnP Modern Search which will include version 4 of MGT that doesn't have this problem with running different versions on the same page.

Also have a look in the Developer Tools Console (Press F12 and go to the Console tab), reload the page and see if anything related to MGT is logged there.

**Reply by calebmil** [reply]
Patrik,

That was the piece I was missing.  Thank you and thank you to Kasper for the template too!

Caleb

**Reply by kasperbolarsen** [reply]
Sorry about that,  I always switch on MGT by default as soon as I add the web part, so I forgot to mention that.
